src/App.jsx

```diff
@@ -324,77 +290,65 @@ function App() {
 
 	function render_in_vr() {
 		renderer.render(scene, camera);
-		if ( vr_touch_down ) {
-			var touch_length = Date.now() - vr_touch_start_time;
-			console.debug( touch_length );
-			if ( touch_length > 3000 ) {
-				console.debug( 'closing session' );
-				console.debug( renderer.xr );
-				console.debug( renderer.xr.getSession() );
-				renderer.xr.getSession()?.end();
-			}
-		}
-	}
-
-	function start_ar() {
-		if (! ('xr' in navigator)) {
-			console.error('xr not supported');
-			alert('xr not supported');
-			return;
-		}
-
-		navigator.xr.isSessionSupported('immersive-ar')
-		.then( (supported) => {
-			if (!supported) {
-				console.error('ar not supported');
-				alert('ar not supported');
-			}
-			else {
-				navigator.xr.requestSession(AR, vr_session_init_options)
-				.then( (session) => {
-					console.debug('ar ready');
-					console.debug(session);
-					current_mode = AR;
-					xr_session = session;
-					init_ar(session);
-				})
-				.catch( (error) => {
-					console.error(error);
-				});
-			}
-		})
 	}
 
 	function start_vr() {
-		create_vr_session( camera, renderer, null, null, () => {
-			start_vol_player();
+		create_vr_session( init_vr, ( error ) => {
+			console.error( error );
 		} );
 	}
 
-	function vr_touch_handler( event ) {
-		console.debug( event );
-		switch (event.type) {
-			case "touchstart": break;
-			case "touchmove": break;
-			case "touchend": break;
-			case "mousedown": vr_touch_start( event ); break;
-			case "mousemove": break;
-			case "mouseup": vr_touch_end( event ); break;
-			default: return;
-		}
+	function init_vr( session ) {
+		camera = new THREE.PerspectiveCamera( 
```

Comment:
if you move this into the render() functions it will resize the view automatically when the window resizes

Reply:
`camera = new THREE.PerspectiveCamera(50, window.innerWidth / window.innerHeight, 0.1, 10 );`